### PR TITLE
Adds mget for Option[T].

### PR DESCRIPTION
**Pull request by dom96**

### CODE REVIEW

**Comment by hlaaftana** [comment]
Would \`[]` work as well?

**Comment by twetzel59** [comment]
@hlaaftana, definitely that would work syntactically, but semantically we should think about it a bit.

[] is meant for dereferencing, so we would not want to confuse a pointer indirection with simply inner field access. For example, in C++ you wouldn't implement the dereference "star" (*) if you didn't have a pointer type.

On the other hand, I really like this idea that you're presenting. It can be argued that it is more orthogonal to use []. Nim can try to be increasingly self-consistent. Using the [] means that access is the same for ``ptr``, ``ref``, and ``Option[T]``.

So, the question is: do we want this similarity between our pointers and our optional type (or sum types in general)?

**Comment by dom96** [comment]
> So, the question is: do we want this similarity between our pointers and our optional type (or sum types in general)?

I would say: no, we do not.

**Comment by zielmicha** [comment]
> For example, in C++ you wouldn't implement the dereference "star" (*) if you didn't have a pointer type.

std::optional<T> actually implements "star" (operator*).

File: lib/pure/options.nim [hunk 1]
```diff
@@ -139,6 +139,13 @@ proc get*[T](self: Option[T], otherwise: T): T =
   else:
     otherwise
 
+proc mget*[T](self: var Option[T]): var T =
+  ## Returns contents of the Option. If it is none, then an exception is
+  ## thrown.
+  if self.isNone:
+    raise UnpackError(msg : "Can't obtain a value from a `none`")
```
Comment (Araq) [inline lib/pure/options.nim]: Bugs should not be mapped to exceptions. This should use ``assert``/``doAssert``.

Comment (twetzel59) [reply lib/pure/options.nim]: Agreed!

Comment (dom96) [reply lib/pure/options.nim]: This is consistent with the rest of ``options``. We can fix it in a separate PR.

**Comment by Bulat-Ziganshin** [comment]
Are I'm only one who dislike 'mget' name? '[]\ is OK for me, otherwise... may be 'value'?

**Comment by dom96** [comment]
We already use ``mget`` in various places https://nim-lang.org/docs/theindex.html#mget

File: lib/pure/options.nim (hunk 1 above)
Comment (Araq) [reply lib/pure/options.nim]: Fair enough.

Comment (Araq) [reply lib/pure/options.nim]: Nitpick: No space before a colon.

**Comment by Araq** [comment]
Yeah, ``mget`` is deprecated. :-( No idea what would be a good name for this.

**Comment by cooldome** [comment]
how about to call it simply ``get``,  it should  ok to have 2 overloads with the same name

**Comment by dom96** [comment]
Renamed to ``get``, can we merge this already?

**Pull request merged**